src/components/workspace-utils.js

```diff
@@ -25,12 +25,12 @@ export const useWorkspaces = ({ persist } = {}) => {
   }
   useEffect(() => {
     refresh()
-  }, [])
+  })
```

Comment:
I don't think we want this change. The empty array causes it to only run once on mount, while omitting it will run on every update.

Reply:
This was to placate the `react-hooks/exhaustive-deps` lint rule. Without removing it: 
> React Hook useEffect has a missing dependency: 'refresh'. Either include it or remove the dependency array

and with it set to `[refresh]`:
> The 'refresh' function makes the dependencies of useEffect Hook (at line 28) change on every render. To fix this, wrap the 'refresh' definition into its own useCallback() Hook

Thoughts?

Reply:
This is a bit complicated. Technically the second is correct, and `refresh` should be wrapped in `useCallback` and specify all _its_ dependencies, etc. Doing all that should result in the behavior we want and satisfy the lint rule. However:
1. That seems like a rather roundabout and fragile way of expressing what we mean, which is 'run on mount'.
2. The hooks docs specifically say that the memoization of `useCallback` is not guaranteed, meaning we could theoretically see spontaneous extra calls in the future.

I see the reason for the lint rule, and it makes sense to be rigorous about the dependencies when we're doing things like data synchronization, where correctness could be impacted. But there are also cases like this where we just want to say "do this once on mount". Perhaps it's worth formally abstracting that idea like so:
```
export const useOnMount = fn => {
  useEffect(fn, []) // eslint-disable-line react-hooks/exhaustive-deps
}
```
(Adding the lint rule and this infrastructure could be its own PR)

Reply:
The [official suggestion](https://reactjs.org/docs/hooks-faq.html#what-can-i-do-if-my-effect-dependencies-change-too-often) for this seems to be `useReducer`, but I prefer this over that because it's simpler and communicates the intention more explicitly.